daemon/daemon.go

```diff
@@ -123,66 +127,114 @@ func (d *Daemon) queueJob(do JobFunc) job.ID {
 		// make a working clone so we don't mess with files we will be
 		// reading from elsewhere
 		Do: func() error {
+			started := time.Now().UTC()
+			d.JobStatusCache.SetStatus(id, job.Status{StatusString: job.StatusRunning})
 			working, err := d.Checkout.WorkingClone()
 			if err != nil {
+				d.JobStatusCache.SetStatus(id, job.Status{StatusString: job.StatusFailed, Err: err.Error()})
 				return err
 			}
 			defer working.Clean()
-			return do(working)
+			metadata, err := do(id, working)
+			if err != nil {
+				d.JobStatusCache.SetStatus(id, job.Status{StatusString: job.StatusFailed, Err: err.Error()})
+				return err
+			}
+			d.JobStatusCache.SetStatus(id, job.Status{StatusString: job.StatusSucceeded, Result: *metadata})
+			var serviceIDs []flux.ServiceID
+			for id, result := range *metadata.Result {
+				if result.Status == update.ReleaseStatusSuccess {
+					serviceIDs = append(serviceIDs, id)
+				}
+			}
+			return d.LogEvent(history.Event{
+				ServiceIDs: serviceIDs,
+				Type:       history.EventCommit,
+				StartedAt:  started,
+				EndedAt:    started,
+				LogLevel:   history.LogLevelInfo,
+				Metadata:   metadata,
+			})
 		},
 	})
+	d.JobStatusCache.SetStatus(id, job.Status{StatusString: job.StatusQueued})
 	println("enqueued job " + id)
 	return id
 }
 
 // Apply the desired changes to the config files
 func (d *Daemon) UpdateManifests(spec update.Spec) (job.ID, error) {
 	switch s := spec.Spec.(type) {
-	case flux.ReleaseSpec:
-		return d.queueJob(func(working git.Checkout) error {
+	case update.ReleaseSpec:
+		return d.queueJob(func(jobID job.ID, working *git.Checkout) (*history.CommitEventMetadata, error) {
 			rc := release.NewReleaseContext(d.Cluster, d.Registry, working)
-			_, err := release.Release(rc, s)
-			return err
+			revision, result, err := release.Release(rc, s)
+			if err == nil {
+				d.askForSync()
+			}
+			return &history.CommitEventMetadata{
+				Revision: revision,
+				Spec:     &spec,
+				Result:   &result,
+			}, nil
 		}), nil
 	case policy.Updates:
-		return d.queueJob(func(working git.Checkout) error {
-			started := time.Now().UTC()
+		return d.queueJob(func(jobID job.ID, working *git.Checkout) (*history.CommitEventMetadata, error) {
 			// For each update
 			var serviceIDs []flux.ServiceID
-			for serviceID, update := range s {
-				serviceIDs = append(serviceIDs, serviceID)
+			metadata := &history.CommitEventMetadata{
+				Spec:   &spec,
+				Result: &update.Result{},
+			}
+			for serviceID, u := range s {
 				// find the service manifest
 				err := d.Cluster.UpdateManifest(working.ManifestDir(), string(serviceID), func(def []byte) ([]byte, error) {
-					return d.Cluster.UpdatePolicies(def, update)
+					newDef, err := d.Cluster.UpdatePolicies(def, u)
+					if err != nil {
+						(*metadata.Result)[serviceID] = update.ServiceResult{
```

Comment:
Might not be needed. I wasn't sure on `omitempty` and go type aliases to maps interaction.